scripts/validate_sdd.py

```
@@ -433,50 +637,253 @@ def validate_invariants(sdd: str, errors: list[str]) -> int:
+    )
+    expected_lineage = [merge_ref, merge_parent_ref]
+    actual_lineage = parents.stdout.strip().split()
+    if parents.returncode != 0 or actual_lineage != expected_lineage:
+        errors.append(
+            "WP0 squash-merge evidence must bind the merge commit and its parent exactly: "
+            f"expected={expected_lineage}, found={actual_lineage}"
+        )
+
+    if fields.get("wp0_exact_main_verification_status") == "passed":
+        verification_ref = fields.get("wp0_exact_main_verification_ref", "")
+        verification_ref_check = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)